torchci/clickhouse_queries/master_commit_red/query.sql

```diff
@@ -1,38 +1,56 @@
 --- This query is used to show the histogram of trunk red commits on HUD metrics page
 --- during a period of time
-WITH all_jobs AS (
+-- Split up the query into multiple CTEs to make it faster.
+with commits as (
+  select
+    push.head_commit.'timestamp' as time,
+    push.head_commit.'id' as sha
+  from
+    push final
+  where
+    push.ref in ('refs/heads/master', 'refs/heads/main')
+    and push.repository.'owner'.'name' = 'pytorch'
+    and push.repository.'name' = 'pytorch'
+    and push.head_commit.'timestamp' >= {startTime: DateTime64(3)}
+    and push.head_commit.'timestamp' < {stopTime: DateTime64(3)}
+),
+all_runs as (
+  select
+    workflow_run.id as id,
+    workflow_run.head_commit.'id' as sha,
+    workflow_run.name as name,
+    commit.time as time
+  from
+    workflow_run final
```

Comment:
Another optimization opportunity:

Do we really need to know when a job is pending? If not, instead of FINAL we can filter on a terminal conclusion

Reply:
I'm going to leave it as is because I don't want to change the results from the old query